continuous-integration/readme.md

```diff
@@ -1,3 +1,235 @@
+  - Keep the build fast. Consider automated test runtime when deciding to pull in dependencies like databases, external services and mock data loading into your test harness. Slow builds often become a bottleneck for dev team's when parallel builds on a CI server aren't an option. Consider adding max timeout limits for lengthy validations to fail fast and maintain high velocity across the team.
+
+- [ ] **Avoid checking in broken builds**
+  - Automated build checks, tests, lint runs, etc should be validated locally before committing your changes to the scm repo. [Test Driven Development](https://martinfowler.com/bliki/TestDrivenDevelopment.html) is a practice dev crews should consider to help identify bugs and failures as early as possible within the development lifecycle.
+- [ ] **Reporting build failures**
+  - If the build step happens to fail then the build pipeline run status should be reported as failed including relevant logs and stacktraces.
+- [ ] **Test Automation Data Dependencies**
+  - Any mocked dataset(s) used for unit and end-to-end integration tests should be checked into the mainline repository. Minimize any external data dependencies with your build process.
+
+## Git Driven Workflow
+
+- [ ] **Build on commit**
+  - Every commit to the baseline repository should trigger the CI pipeline to create a new build candidate.
+  - Build artifact(s) are built, packaged, validated and deployed continuously into a non-production environment per commit. Each commit against the repository results into a CI run which checks out the sources onto the integration machine, initiates a build, and notifies the committer of the result of the build.
+
+- [ ] **Master branch consistency with production**
+  - Merges into the master branch trigger releases into the production environment(s). This way the master branch becomes a dependable baseline for the code running in production.
+
+- [ ] **Avoid commenting out failing tests**
+  - Avoid commenting out tests in the mainline branch. By commenting out tests, we get an incorrect indication of the status of the build.
+
+- [ ] **Branch policy enforcement**
+  - Branch policies should be setup on the master branch so that the build pipeline status becomes a pre-req validation prior to starting a code review. Code review approvers will only start reviewing a pull request once the CI pipeline run passes for the latest pushed git commit.
+  - Broken builds should block pull request reviews.
+
+## Deliver Quickly and Daily
+
+"By committing regularly, every committer can reduce the number of conflicting changes. Checking in a week's worth of work runs the risk of conflicting with other features and can be very difficult to resolve. Early, small conflicts in an area of the system cause team members to communicate about the change they are making."
```

Comment:
Is this a quote? Consider using the `>` prior to the content to make it a block quote.

> example. this is proceeded by a > character